public/js/components/Editor.js

```diff
@@ -173,14 +186,30 @@ const Editor = React.createClass({
       this.setSourceText(nextProps.sourceText, this.props.sourceText);
     }
 
-    if (this.props.selectedSource && !nextProps.selectedSource) {
+    if (this.props.location && !nextProps.location) {
       this.editor.setText("");
     }
 
-    // Highlight the paused line if necessary
-    if (isSourceForFrame(nextProps.selectedSource, nextProps.selectedFrame)) {
+    // Highlight the paused line if necessary.
+    if (isSourceForFrame(nextProps.location, nextProps.selectedFrame)) {
       this.setDebugLine(nextProps.selectedFrame.location.line);
     }
+
+    // If the location has changed and a specific line is requested,
+    // move to that line.
+    if (this.props.location !== nextProps.location &&
+        nextProps.location &&
+        nextProps.location.line != undefined) {
+      if (this.props.location) {
+        this.clearHighlightLine(this.props.location.line);
+      }
+      this.clearHighlightLine(nextProps.location.line);
+
+      const line = nextProps.location.line;
+      this.editor.alignLine(line);
+      this.setHighlightLine(nextProps.location.line);
+      this.editor.codeMirror.addLineClass(line - 1, "line", "highlight-line");
+    }
```

Comment:
This looks like a good function that we could write unit tests for. 

We don't have a great setup now for writing component unit tests, but it's not terrible either. If you did write them, it would be possible to run them with the `mocha-server` and test passing different props down.